src/streaming-ingestion/parameter.ts

```diff
@@ -0,0 +1,305 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { CfnParameter, CfnRule, Fn } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import {
+  REDSHIFT_CLUSTER_IDENTIFIER_PATTERN,
+  REDSHIFT_DB_USER_NAME_PATTERN,
+} from '../common/constant';
+import { REDSHIFT_MODE } from '../common/model';
+import { Parameters } from '../common/parameters';
+
+export interface StreamingIngestionRedshiftStackProps {
+  projectId: string;
+  appIds: string;
+  streamingIngestionConfiguration: {
+    streamingIngestionRoleArn: string;
+    kinesis?: {
+      kinesisDataStreamNames: string;
+    };
+    msk?: {
+      mskClusterArn: string;
+      mskTopicNames: string;
+    };
+  };
+  redshift: {
+    mode: string;
+    defaultDatabaseName: string;
+    existingServerless?: {
+      workgroupName: string;
+      workgroupId?: string;
+      namespaceId?: string;
+      iamRole: string;
+    };
+    provisioned?: {
+      clusterIdentifier: string;
+      dbUser: string;
+      iamRole: string;
+    };
+  };
+}
+
+export function createStackParameters(scope: Construct): {
+  metadata: {
+    [key: string]: any;
+  };
+  params: StreamingIngestionRedshiftStackProps;
+} {
+  const kinesisDataStreamNamesParam = new CfnParameter(scope, 'KinesisDataStreamNames', {
+    description: 'The names of multiple Kinesis Data Stream used for streaming ingestion, names separated by commas.',
+    type: 'String',
+  });
+
+  const streamingIngestionAssociateRoleParam = new CfnParameter(scope, 'StreamingIngestionAssociateRole', {
+    description: 'The ARN of IAM role of Redshift streaming ingestion.',
+    type: 'String',
+    default: '',
+    allowedPattern: '^(arn:(aws|aws-cn):iam::[0-9]{12}:role/.*)?$',
+  });
+
+  const streamingIngestionConfigurationParamsGroup = [];
+  streamingIngestionConfigurationParamsGroup.push({
+    Label: { default: 'Kinesis Data Stream' },
+    Parameters: [
+      kinesisDataStreamNamesParam.logicalId,
+      streamingIngestionAssociateRoleParam.logicalId,
+    ],
+  });
+
+  const streamingIngestionConfigurationParamsLabels = {
+    [kinesisDataStreamNamesParam.logicalId]: {
+      default: 'The names of multiple Kinesis Data Stream',
+    },
+    [streamingIngestionAssociateRoleParam.logicalId]: {
+      default: 'The ARN of IAM role of Redshift streaming ingestion',
+    },
+  };
+
+
+  const redshiftModeParam = new CfnParameter(scope, 'RedshiftMode', {
```

Comment:
We should share those parameters with data modeling stack. Let's move them to common parameter in helper methods to share the code.

Reply:
done